### PR TITLE
Update ESP32 base samples to work with ESP32 package 3.0.1 (+ Arduino IDE 2.3.2)

### PR DESCRIPTION
**Pull request by ewertons**
This is to address issue https://github.com/Azure/azure-sdk-for-c/issues/2836

### CODE REVIEW

File: examples/Azure_IoT_Central_ESP32/Azure_IoT_Central_ESP32.ino [hunk 1]
```diff
@@ -127,25 +132,48 @@ static int mqtt_client_init_function(
...
-  mqtt_config.event_handle = esp_mqtt_event_handler;
-  mqtt_config.user_context = NULL;
-  mqtt_config.cert_pem = (const char*)ca_pem;
+#if defined(ESP_ARDUINO_VERSION_MAJOR) && ESP_ARDUINO_VERSION_MAJOR >= 3
+    mqtt_config.broker.address.uri = mqtt_broker_uri;
+    mqtt_config.broker.address.port = mqtt_client_config->port;
+    mqtt_config.credentials.client_id = (const char*)az_span_ptr(mqtt_client_config->client_id);
+    mqtt_config.credentials.username = (const char*)az_span_ptr(mqtt_client_config->username);
+
+  #ifdef IOT_CONFIG_USE_X509_CERT
```
Comment (RLeclair) [inline examples/Azure_IoT_Central_ESP32/Azure_IoT_Central_ESP32.ino:141]: Nit: Do nested `#ifdef`s need this spacing?

File: examples/Azure_IoT_Central_ESP32/Azure_IoT_Central_ESP32.ino [hunk 2]
```diff
@@ -611,7 +652,10 @@ static esp_err_t esp_mqtt_event_handler(esp_mqtt_event_handle_t event)
       break;
   }
 
-  return ESP_OK;
+#if defined(ESP_ARDUINO_VERSION_MAJOR) && ESP_ARDUINO_VERSION_MAJOR >= 3
+#else // ESP_ARDUINO_VERSION_MAJOR
+return ESP_OK;
```
Comment (RLeclair) [inline examples/Azure_IoT_Central_ESP32/Azure_IoT_Central_ESP32.ino]: Nit:
```suggestion
  return ESP_OK;
```

**Review by RLeclair — COMMENTED** [review]
LGTM!

**Review by RLeclair — APPROVED** [review]

File: examples/Azure_IoT_Central_ESP32/Azure_IoT_Central_ESP32.ino (hunk 1 above)
Comment (ewertons) [reply examples/Azure_IoT_Central_ESP32/Azure_IoT_Central_ESP32.ino:141]: No they do not, I just did that to improve readability.

**Pull request merged**